app/src/test/java/com/losandroides/learn/presentation/MainViewModelTest.kt

```diff
@@ -20,16 +21,32 @@ class MainViewModelTest {
     private val itemsUseCase = mockk<ItemsUseCase>()
 
     @Test
-    fun `GIVEN some items WHEN MainViewModel is initialized THEN ItemsUseCase is called`() =
+    fun `GIVEN some items WHEN MainViewModel is initialized THEN ItemsUseCase is called AND has correct state order (LOADING - CONTENT)`() =
+        runTest(testDispatcher) {
+            coEvery { itemsUseCase() } returns Throwable("Something happened").left()
+
+            val viewModel = buildMainViewModel()
+
+            viewModel.viewState.value shouldBeEqualTo MainViewModel.ViewState.Loading
+            runCurrent()
+            viewModel.viewState.value shouldBeEqualTo MainViewModel.ViewState.Error
+
+            coVerifyOnce { itemsUseCase() }
+        }
+
+    @Test
+    fun `GIVEN error WHEN MainViewModel is initialized THEN correct state order (LOADING - ERROR)`() =
```

Comment:
This is triggering a detekt issue because of the length of the line.

Reply:
I'd propose other way of doing the naming of the test, have you heard about [`@DisplayName()`](https://junit.org/junit5/docs/5.0.3/api/org/junit/jupiter/api/DisplayName.html)?
I'm used to use this it would be something like this : 
```
@Test
@DisplayName("""
GIVEN error
WHEN mainViewModel is initialised
THEN flow has the correct state order (LOADING - ERROR)
""")
fun test() {...}
```

So detekt will never complay about test that has a large gerking.

Reply:
In addition, we can relax some rules for the testing classes.

Reply:
@RubenGonz2  what do you mean with "relax some rules"?

Reply:
I mean we can ignore some of the detekt rules. Some of them doesn't make sense in testing classes.

We can add exclusions for the Tests classes.

You can find examples in the config file: config/detekt/detekt.yml

We need to add something like this in every rule we want to exclude:


```
excludes: ['**/test/**', '**/androidTest/**', '**/commonTest/**', '**/jvmTest/**', '**/jsTest/**', '**/iosTest/**']
```

Reply:
I think it is a good deal to relax the test rules.

And on the other hand, my mind blew up with the `DisplayName` thingy.
I didn't know it!! :rocket:

Reply:
Ammmm it is on Junit 5. I didn't start yet with it. Maybe it is a good moment! :blush:

Reply:
Sure, I can create a new PR updating to JUnit5, update everything and then merge this according with the new JUnit5, so we don't change/alter any detekt rules, what do you think @soygabimoreno @RubenGonz2

Reply:
Amazing!